tenseal/cpp/tensors/encrypted_vector.h

```diff
@@ -69,78 +68,229 @@ class EncryptedVector
         return this->copy()->replicate_first_slot_inplace(n);
     }
     virtual encrypted_t replicate_first_slot_inplace(size_t n) = 0;
-    void broadcast_or_throw(encrypted_t other) {
-        if (this->size() == other->size()) return;
+    /**
+     * Adjust two vectors to match sizes.
+     * @return the right operand, in case it was copied and altered.
+     **/
+    encrypted_t broadcast_or_throw(encrypted_t other) {
```

Comment:
Would it help to pass `other` by reference since it's not modified?

Reply:
Ah sorry, I guess encrypted_t is already a pointer

Reply:
Yeah, I should use const reference everywhere, in case encrypted_t is not a pointer